### ISSUE TITLE
Allow large entities: Large collision boxes, large selection boxes, large visuals

### ISSUE THREAD

**Issue by appgurueu** [minetest/minetest#14686]
### Problem

Large entities don't work well. Minetest operates under the (pretty much hard-coded) assumption that all entities are "reasonably small" which is somewhere around a radius of a few nodes (?):

* With a viewing range of 100, a 20 node large entity with a center which is 100 nodes away may become invisible - but it should be visible.
* Large collision boxes don't work.
* Large selection boxes don't work.

(For use cases, consider e.g. a boss, a dragon, a blue whale...)

### Solutions

The "cheap" solution would be to just bump margins everywhere. This need not even be that bad. But it doesn't scale well, because the larger the margins, the worse the performance is.

A better solution is probably to bucket entities by their collision / selection box size. For small entities, you can still use small margins; only the few big entities will then require bigger margins.

### Alternatives

To an extent, modders can implement their own janky collisions or selections. But this isn't really viable. Practically modders end up being forced to downsize their entities.

**Comment by nininik0** [comment]
i agree. i have a giant unkown object boss as a server mob, and it dissapears.

**Comment by Desour** [comment]
Related: #8381

----

For very large entities, we also need to send them to players from larger distances, and maybe not store them in mapblocks.

**Comment by nininik0** [comment]
I'm kinda noob at stuff I'm sorta good at lua but maybe we can make it so that in order for it to not disappear we can make the dissappear distance bigger by option, like in minetest.conf it could go like: large_entity_render_range = true
